lizmap/modules/lizmap/lib/Logger/Config.php

```diff
@@ -0,0 +1,202 @@
+<?php
+/**
+ * Manage and give access to lizmap log configuration.
+ *
+ * @author    3liz
+ * @copyright 2020 3liz
+ *
+ * @see      http://3liz.com
+ *
+ * @license Mozilla Public License : http://www.mozilla.org/MPL/
+ */
+
+namespace Lizmap\Logger;
+
+use Lizmap\App;
+
+class Config
+{
+    // Lizmap log configuration data
+    private $data = array();
+
+    // general properties
+    private $properties = array(
+        'active',
+        'profile',
+    );
+
+    /**
+     * @var array list of the logItems of the ini file
+     */
+    protected $logItems = array();
+
+    /**
+     * @var string If the log is active globally or not
+     */
+    private $active = '';
+
+    /**
+     * @var string database profile
+     */
+    private $profile = '';
+
+    /**
+     * @var string The location of the ini file
+     */
+    protected $file;
+
+    /**
+     * @var App\AppContextInterface
+     */
+    protected $appContext;
+
+    /**
+     * Constructs the object. This method shouldn't be called, you should call lizmap::getLogConfig() instead.
+     *
+     * @param array                   $configData  An array containing the ini file
+     * @param App\AppContextInterface $appContext  The context
+     * @param string                  $iniFilePath the path to the ini file folder
+     */
+    public function __construct($configData, App\AppContextInterface $appContext, $iniFilePath)
+    {
+        $this->data = $configData;
+        $this->appContext = $appContext;
+        $this->file = $iniFilePath;
+
+        // set generic parameters
+        foreach ($this->properties as $prop) {
+            if (isset($configData['general'][$prop])) {
+                $this->{$prop} = $configData['general'][$prop];
+            }
+        }
+    }
+
+    public function getProperties()
+    {
+        return $this->properties;
+    }
+
+    /**
+     * Get a log item.
+     *
+     * @param string $key Key of the log item to get
+     *
+     * @return Log\Item
+     */
+    public function getLogItem($key)
+    {
+        if (!key_exists($key, $this->logItems)) {
+            if (!key_exists('item:'.$key, $this->data)) {
+                return null;
+            }
+            $this->logItems[$key] = new Item($key, $this->data['item:'.$key], $this->appContext);
+        }
+
+        return $this->logItems[$key];
+    }
+
+    /**
+     * Get a list of log items names.
+     *
+     * @return string[] list of names
+     */
+    public function getLogItemList()
+    {
+        $logItemList = array();
+
+        foreach ($this->data as $section => $data) {
+            if (strstr($section, 'item:')) {
+                $logItemList[] = preg_replace('#(^item:)#', '', $section, 1);
```

Comment:
Not optimized. Your code search two times in the string. 

Here is a better solution, search and retrieve at the same time:

```
           if (preg_match('/^item:(.*)/', $section, $matches)) {
                  $logItemList[] = $matches[1];
           }
```

Reply:
I'll fix it